### PR TITLE
[LocalNews8] add extractor (Closes #9200)

### LINKED ISSUE #9200: Support localnews8.com
## Please follow the guide below

- You will be asked some questions and requested to provide some information, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *issue* (like that [x])
- Use *Preview* tab to see how your issue will actually look like

---

### Make sure you are using the *latest* version: run `youtube-dl --version` and ensure your version is *2016.04.13*. If it's not read [this FAQ entry](https://github.com/rg3/youtube-dl/blob/master/README.md#how-do-i-update-youtube-dl) and update. Issues with outdated version will be rejected.
- [x] I've **verified** and **I assure** that I'm running youtube-dl **2016.04.13**

### Before submitting an *issue* make sure you have:
- [x] At least skimmed through [README](https://github.com/rg3/youtube-dl/blob/master/README.md) and **most notably** [FAQ](https://github.com/rg3/youtube-dl#faq) and [BUGS](https://github.com/rg3/youtube-dl#bugs) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?type=Issues) the bugtracker for similar issues including closed ones

### What is the purpose of your *issue*?
- [ ] Bug report (encountered problems with youtube-dl)
- [x] Site support request (request for adding support for a new site)
- [ ] Feature request (request for a new functionality)
- [ ] Question
- [ ] Other

---

### The following sections concretize particular purposed issues, you can erase any section (the contents between triple ---) not applicable to your *issue*

---

### If the purpose of this *issue* is a *bug report*, *site support request* or you are not completely sure provide the full verbose output as follows:

Add `-v` flag to **your command line** you run youtube-dl with, copy the **whole** output and insert it here. It should look similar to one below (replace it with **your** log inserted between triple ```):
```
$ youtube-dl -v http://www.localnews8.com/news/rexburg-business-turns-carbon-fiber-scraps-into-wedding-rings/35183304
[debug] System config: []
[debug] User config: [u'--prefer-ffmpeg', u'--download-archive', u'/home/ben/var/youtube-dl-archive', u'-i', u'--write-description', u'--write-info-json', u'--write-annotations', u'--write-sub', u'--output', u'%(title)s-%(extractor_key)s-%(id)s.%(ext)s']
[debug] Command-line args: [u'-v', u'http://www.localnews8.com/news/rexburg-business-turns-carbon-fiber-scraps-into-wedding-rings/35183304']
[debug] Encodings: locale UTF-8, fs UTF-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2016.04.13
[debug] Python version 2.7.11+ - Linux-4.4.0-1-amd64-x86_64-with-debian-stretch-sid
[debug] exe versions: avconv 2.8.6-1, avprobe 2.8.6-1, ffmpeg 2.8.6-1, ffprobe 2.8.6-1, rtmpdump 2.4
[debug] Proxy map: {}
[generic] 35183304: Requesting header
WARNING: Falling back on generic information extractor.
[generic] 35183304: Downloading webpage
[generic] 35183304: Extracting information
ERROR: Unsupported URL: http://www.localnews8.com/news/rexburg-business-turns-carbon-fiber-scraps-into-wedding-rings/35183304
Traceback (most recent call last):
  File "/home/ben/bin/youtube-dl/youtube_dl/extractor/generic.py", line 1360, in _real_extract
    doc = compat_etree_fromstring(webpage.encode('utf-8'))
  File "/home/ben/bin/youtube-dl/youtube_dl/compat.py", line 279, in compat_etree_fromstring
    doc = _XML(text, parser=etree.XMLParser(target=etree.TreeBuilder(element_factory=_element_factory)))
  File "/home/ben/bin/youtube-dl/youtube_dl/compat.py", line 268, in _XML
    parser.feed(text)
  File "/usr/lib/python2.7/xml/etree/ElementTree.py", line 1653, in feed
    self._raiseerror(v)
  File "/usr/lib/python2.7/xml/etree/ElementTree.py", line 1517, in _raiseerror
    raise err
ParseError: not well-formed (invalid token): line 3, column 127
Traceback (most recent call last):
  File "/home/ben/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 671, in extract_info
    ie_result = ie.extract(url)
  File "/home/ben/bin/youtube-dl/youtube_dl/extractor/common.py", line 341, in extract
    return self._real_extract(url)
  File "/home/ben/bin/youtube-dl/youtube_dl/extractor/generic.py", line 2024, in _real_extract
    raise UnsupportedError(url)
UnsupportedError: Unsupported URL: http://www.localnews8.com/news/rexburg-business-turns-carbon-fiber-scraps-into-wedding-rings/35183304
```

---

### If the purpose of this *issue* is a *site support request* please provide all kinds of example URLs support for which should be included (replace following example URLs by **yours**):
- Single video (original containing page): http://www.localnews8.com/news/rexburg-business-turns-carbon-fiber-scraps-into-wedding-rings/35183304
- Single video (actual URL - already works via Generic but gives horrible id): http://cdnapi.kaltura.com/p/557781/sp/55778100/playManifest/entryId/1_ubaezv9y/flavorId/1_dy8kcmjj/format/url/protocol/http/a.mp4?referrer=aHR0cDovL3d3dy5sb2NhbG5ld3M4LmNvbQ==&playSessionId=911bdf61-144e-f90d-d8c6-b031ea06861e&clientTag=html5:v2.36&deliveryCode=149353&uiConfId=13089312

---

### Description of your *issue*, suggested solution and other information

There are 2 parts:

* First, you need to enter the iframe, which is created by JS somehow.
* Then, you need to extract the ID properly.

PS, thanks for all your hard work, it makes life with a slow internet connection much more bearable.

### PR DESCRIPTION
**Pull request by TRox1972**

### CODE REVIEW

File: youtube_dl/extractor/localnews8.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+            'timestamp': '1441844822',
+            'uploader_id': 'api',
+        }}
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+
+        partner_id = re.search(r'partnerId\s*:\s*"([0-9]{6}?)"', webpage).group(1)
+        kaltura_id = re.search(r'var\s*videoIdString\s*=\s*"kaltura:(.+?)";', webpage).group(1)
```
Comment (yan12125) [inline youtube_dl/extractor/localnews8.py]: Use ```self._search_regex``` instead.

**Comment by TRox1972** [comment]
Thanks for feedback @yan12125. Does this seem good?

File: youtube_dl/extractor/localnews8.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+
+
+class LocalNews8IE(InfoExtractor):
+    _VALID_URL = r'http?://(?:www\.)?localnews8\.com/.+?/(?P<id>[0-9]+)'
```
Comment (dstftw) [inline youtube_dl/extractor/localnews8.py]: It should be `https?`.

File: youtube_dl/extractor/localnews8.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+            'duration': '153',
+            'timestamp': '1441844822',
+            'uploader_id': 'api',
+        }}
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+
+        partner_id = self._search_regex(r'partnerId\s*:\s*"([0-9]{6}?)"', webpage, video_id)
```
Comment (dstftw) [inline youtube_dl/extractor/localnews8.py]: Change group to `(\d+)`.

File: youtube_dl/extractor/localnews8.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+        kaltura_id = self._search_regex(r'var\s*videoIdString\s*=\s*"kaltura:(.+?)";', webpage, video_id)
```
Comment (dstftw) [inline youtube_dl/extractor/localnews8.py]: Should be `var\s+`.

File: youtube_dl/extractor/localnews8.py (hunk 2 above)
Comment (TRox1972) [inline youtube_dl/extractor/localnews8.py]: OK, I've changed it. But as of now, accessing a site on the webpage using https results in a SSL certificate error or, with `--no-check-certificates`, a 503 service unavailable. Should I add a statement that catches these errors and change the protocol to http?

**Pull request closed without merging**